### ISSUE TITLE
Chapter 2.2 TRACEABILITY

### ISSUE THREAD

**Issue by bms63** [pharmaverse/admiral-adamig-wg#16]

**Comment by bms63** [comment]
I don't think we produce any "intermediate datasets" for other ADaMS to consume???  Maybe the Hy's Law dataset is a candidate as it consumes ADLB.

Wondering if we need an example of this in the documentationprobably not a big deal to not include

**Comment by cpiraux** [comment]
My understanding of intermediate datasets is that they are created primarily to support traceability and are not necessarily used as input datasets for displays. I think an intermediate dataset for intercurrent events could be a useful example, as I find traceability challenging for ICE.

**Comment by bms63** [comment]
oh yeah ICE is a hot topic and makes my brain hurt.

The CDISC Pilot study wasn't set up around estimand strategies - not sure if we should come up with our own!

**Comment by bms63** [comment]
Would be awesome!!  Keep on the backlog

**Comment by bms63** [comment]
Opportunity to work with Pilot Working Groups that are building larger and modern datasets for submission to the FDA

**Comment by vfinneme** [comment]
There is not a lot of documentation on traceability variables in the admiral guidance.  In many of the vignettes, the original SDTM --SEQ variables are carried forward into the generated ADaM dataset without fanfare.  Indeed, the purpose of the creation of ASEQ is usually to support incorporation into a subsequent dataset and there are a number of examples that demonstrate how to create the ASEQ variable.  

The only places SRCDOM/SRCVAR/SRCSEQ are mentioned is within the context of time-to-event datasets.  While they appear to be correctly implemented, it would be useful to demonstrate implementation in other datasets (e.g. in the case where source records for newly derived PARAMs are not retained or when an intermediate ADaM is used and the resultant dataset is NOT TTE).  A good use case where the implementation would be a little bit more confusing is merging in records from multiple source datasets and retaining traceability from each.  Based on the TTE examples, this should be as easy as adding a set_values_to option to the appropriate function within a vignette.  This is technically exactly what is demonstrated in the TTE examples, but would add clarity that this can be implemented in a variety of the admiral functions and workflows.

**Comment by vfinneme** [comment]
Regarding earlier comments on ICEs, I suspect that the sensible flow here is to wait until an admiral module is design/built specifically with ICEs in mind.

**Comment by bms63** [comment]
Maybe a good topic for next week's meeting - some pseudo code and data for making an ICE module for admiral.

**Comment by adcascone** [comment]
Discussion 18 February 

srcvar definition -> needs to point to the variable that's used to create AVAL 
it's close here: **https://pharmaverse.github.io/admiral/cran-release/articles/bds_tte.html?q=time%20to%20event#add-additional-information-for-events-and-censoring-evntdesc-srcvar**

<img width="589" height="810" alt="Image" src="https://github.com/user-attachments/assets/f2482f73-f4c0-48ad-b7e8-8122bee83686" />

BUT AVAL is not the event here, it's the time to the event date, so srcvar should point to something that contains the duration, not the event date 

solution? in the dataset that's you're pulling from, make sure there's a study date variable in there 
if you have a study day variable in the source dataset, you can use that instead of the date for srcvar 

seems like date is used often as scrvar in time-to-event examples but that is not really correct according to the ADaMIG

**Comment by adcascone** [comment]
Assigning ASEQ - no required way to generate ASEQ, it just needs to be unique 
https://pharmaverse.github.io/admiral/cran-release/articles/bds_exposure.html?q=aseq#aseq

ASEQ is not required in all ADaM datasets, but it's often used that way 

Also discussed in #30

**Comment by bms63** [comment]
https://github.com/pharmaverse/admiral/issues/2986

**Comment by bms63** [comment]
Also note that ICE estimands was created - https://pharmaverse.github.io/admiral/main/articles/estimands.html
